engine/api/worker/model.go

```diff
@@ -50,7 +50,6 @@ func InsertWorkerModel(db gorp.SqlExecutor, model *sdk.Model) error {
 // UpdateWorkerModel update a worker model
 func UpdateWorkerModel(db gorp.SqlExecutor, model sdk.Model) error {
 	model.UserLastModified = time.Now()
-	model.NeedRegistration = true
```

Comment:
needRegistration is always true here. See review of 
```go
func (h *HatcheryVSphere) SpawnWorker
```
I'm not sure that all work on NeedRegistration is usefull.  See that irl :)